src/nullablevector.jl

```diff
@@ -195,7 +186,29 @@ function Base.append!(X::NullableVector, items::AbstractVector)
     old_length = length(X)
     nitems = length(items)
     resize!(X, old_length + nitems)
-    X[old_length + 1:end] = items[1:nitems]
+    copy!(X, length(X)-nitems+1, items, 1, nitems)
+    return X
+end
+
+@doc """
+`prepend!(X::NullableVector, items::AbstractVector)`
+
+Add the elements of `items` to the beginning of `X`.
+
+Note that `prepend!(X, [1, 2, 3])` is equivalent to `unshift!(X, 1, 2, 3)`,
+where the items to be added to `X` are passed individually to `unshift!` and as a
+collection to `prepend!`.
+""" ->
+function Base.prepend!(X::NullableVector, items::AbstractVector)
+    old_length = length(X)
+    nitems = length(items)
+    ccall(:jl_array_grow_beg, Void, (Any, UInt), X.values, nitems)
```

Comment:
This sounds like a feature which should be supported in Julia, for example via `resize!(X, n, beginning::Bool=false)`. For now it's OK to use `ccall`, but would you be willing to make a PR against Julia to add this?